Sources/Fuzzilli/FuzzIL/Operations.swift

```diff
@@ -402,44 +410,60 @@ class BeginArrowFunctionDefinition: BeginAnyFunctionDefinition {}
 class EndArrowFunctionDefinition: EndAnyFunctionDefinition {}
 
 // A ES6 generator function
-class BeginGeneratorFunctionDefinition: BeginAnyFunctionDefinition {}
+class BeginGeneratorFunctionDefinition: BeginAnyFunctionDefinition {
+    init(signature: FunctionSignature, isStrict: Bool) {
+        super.init(signature: signature, isStrict: isStrict, contextOpened: [.script, .function, .generatorFunction])
+    }
+}
 class EndGeneratorFunctionDefinition: EndAnyFunctionDefinition {}
 
 // A ES6 async function
-class BeginAsyncFunctionDefinition: BeginAnyFunctionDefinition {}
+class BeginAsyncFunctionDefinition: BeginAnyFunctionDefinition {
+    init(signature: FunctionSignature, isStrict: Bool) {
+        super.init(signature: signature, isStrict: isStrict, contextOpened: [.script, .function, .asyncFunction])
+    }
+}
 class EndAsyncFunctionDefinition: EndAnyFunctionDefinition {}
 
 // A ES6 async arrow function
-class BeginAsyncArrowFunctionDefinition: BeginAnyFunctionDefinition {}
+class BeginAsyncArrowFunctionDefinition: BeginAnyFunctionDefinition {
+    init(signature: FunctionSignature, isStrict: Bool) {
+        super.init(signature: signature, isStrict: isStrict, contextOpened: [.script, .function, .asyncFunction])
+    }
+}
 class EndAsyncArrowFunctionDefinition: EndAnyFunctionDefinition {}
 
 // A ES6 async generator function
-class BeginAsyncGeneratorFunctionDefinition: BeginAnyFunctionDefinition {}
+class BeginAsyncGeneratorFunctionDefinition: BeginAnyFunctionDefinition {
+    init(signature: FunctionSignature, isStrict: Bool) {
+        super.init(signature: signature, isStrict: isStrict, contextOpened: [.script, .function, .asyncFunction, .generatorFunction])
+    }
+}
 class EndAsyncGeneratorFunctionDefinition: EndAnyFunctionDefinition {}
 
 class Return: Operation {
     init() {
-        super.init(numInputs: 1, numOutputs: 0, attributes: [.isJump])
+        super.init(numInputs: 1, numOutputs: 0, attributes: [.isJump], requiredContext: .function)
```

Comment:
I think all these non-default `requiredContext` should also include `.script`, since that's the default value (line 47), and they do require `.script` (it's just that there is currently no case where there'd be `.function` context and not also `.script` context). WDYT?

Reply:
Yep, sounds good! I'll add this context.